### PR TITLE
Convert an MRT archive of updates into BigQuery-friendly data

### PR DESCRIPTION
**Pull request by yuhao-he**

### CODE REVIEW

File: server/mrt_converter/mrt_converter.go [hunk 1]
```diff
@@ -112,3 +118,56 @@ func parseUpdate(collector string, h *mrt.MRTHeader, buf []byte) (*update, error
...
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return fmt.Errorf("failed to read MRT header: %v", err)
+		}
+
+		h := &mrt.MRTHeader{}
+		err = h.DecodeFromBytes(buf)
+		if err != nil {
+			return fmt.Errorf("DecodeFromBytes: %v", err)
```
Comment (morrowc) [inline server/mrt_converter/mrt_converter.go:141]: some golang style nerd is going to say:
  "Hey, errors shouldnt' strart with capitalization nor end in punctuation"
  
  you might:
    "failed to decode header from bytes: %v"

instead :) or not, your call :)

File: server/mrt_converter/mrt_converter.go [hunk 2]
```diff
@@ -112,3 +118,56 @@ func parseUpdate(collector string, h *mrt.MRTHeader, buf []byte) (*update, error
 		Attributes: translateAttrs(bgpUpdate.PathAttributes),
 	}, nil
 }
+
+// convert translates the bzip'ed MRT raw bytes into a BigQuery compatible
+// format and write to GCS.
+func convert(collector string, src []byte, dst io.Writer) error {
+	br := bzip2Reader(bytes.NewReader(src))
+	gr := gzip.NewWriter(dst)
```
Comment (morrowc) [inline server/mrt_converter/mrt_converter.go:126]: this might make more sense as 'gw', but 'gr' is probably also ok:)
style/etc call for you to decide on.

**Review by morrowc — APPROVED** [review]
A couple style based things, nothing significant :)

File: server/mrt_converter/mrt_converter.go (hunk 1 above)
Comment (yuhao-he) [reply server/mrt_converter/mrt_converter.go]: Oops, didn't notice it. 

Done.

File: server/mrt_converter/mrt_converter.go (hunk 2 above)
Comment (yuhao-he) [reply server/mrt_converter/mrt_converter.go]: Yes - "gw" is a better name. 

btw, I got some pings for internal changelists, and someone told me not to use global var any more, so I use dependency injection here instead.

**Comment by morrowc** [comment]
On Tue, 07 Dec 2021 06:38:46 +0000,
yuhao-he ***@***.***> wrote:
> 
> [1  <text/plain; UTF-8 (7bit)>]
> @yuhao-he commented on this pull request.
> 
> 
> 
> > @@ -112,3 +118,56 @@ func parseUpdate(collector string, h *mrt.MRTHeader, buf []byte) (*update, error
>  		Attributes: translateAttrs(bgpUpdate.PathAttributes),
>  	}, nil
>  }
> +
> +// convert translates the bzip'ed MRT raw bytes into a BigQuery compatible
> +// format and write to GCS.
> +func convert(collector string, src []byte, dst io.Writer) error {
> +	br := bzip2Reader(bytes.NewReader(src))
> +	gr := gzip.NewWriter(dst)
> 
> Yes - "gw" is a better name. 
> 
> btw, I got some pings for internal changelists, and someone told me not to use global var any more, so I use dependency injection here instead.

I saw some notice on using dependency injection to avoid global vars
in testing/production use-cases (in my travels to understand
dependency injection better)

**Comment by morrowc** [comment]
On Tue, 07 Dec 2021 06:39:21 +0000,
yuhao-he ***@***.***> wrote:
> 
> [1  <text/plain; UTF-8 (7bit)>]
> Merged #3 into main.

I think there's 1 set of things in a PR to merge.  Let me know when
you're all closed out (no comments/questions from me on the pending)
and I'll click the 'merge' button.

Of course, you MAY be able to click that button yourself?
if so, please do! :)

**Pull request merged**